msu/hooks/skills/skill_container.nut

```diff
@@ -27,6 +27,74 @@
 		this.m.ScheduledChangesSkills.clear();
 	}
 
+	local add = o.add;
+	o.add = function( _skill, _order = 0 )
+	{
+		if (!_skill.isStacking())
+		{
+			local incrementAddedStack = function( _alreadyPresentSkill, _skillToAdd )
+			{
+				_alreadyPresentSkill.m.MSU_AddedStack += 1;
+				if (!::MSU.isNull(_skillToAdd.getItem()))
+				{
+					_alreadyPresentSkill.setItem(_skillToAdd.getItem());
+					foreach (i, itemSkill in _skillToAdd.getItem().m.SkillPtrs)
+					{
+						if (itemSkill.getID() == _skillToAdd.getID())
+						{
+							_skillToAdd.getItem().m.SkillPtrs[i] = _alreadyPresentSkill;
+							_skillToAdd.setItem(null);
+							break;
+						}
+					}
+				}
+
+				if (_alreadyPresentSkill.m.MSU_AddedStack > 0)
+				{
+					_alreadyPresentSkill.m.IsGarbage = false;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.Skills)
+			{
+				if (alreadyPresentSkill.getID() == _skill.getID())
+				{
+					incrementAddedStack(alreadyPresentSkill, _skill);
+					if (alreadyPresentSkill.m.MSU_AddedStack > 1) alreadyPresentSkill.onRefresh();
+					return;
+				}
+			}
+
+			foreach (i, alreadyPresentSkill in this.m.SkillsToAdd)
+			{
+				if (alreadyPresentSkill.getID() == _skill.getID())
+				{
+					incrementAddedStack(alreadyPresentSkill, _skill);
+					return;
+				}
+			}
+		}
```

Comment:
I think we should prioritize keeping the skill and item linked as much as possible, and only switch to a different item if the one currently linked is removed, or if the skill wasn't linked to an item in the first place.

Right now we swap items every time a new item is added with this skill.

Idk, I think either approach could cause issues but I feel like sticking with the same item is safer.

Reply:
That approach is what I went with initially but it makes the entire thing **extremely** messy. Because items hold references directly to the skill objects, not ids. So we'd have to modify the `this.m.SkillPtrs` array of the new item and find the skill, remove it, and add the old skill. This also causes issues if the old skill was referenced elsewhere in the code. In this second iteration, it is much cleaner.

Reply:
Why would that be necessary? Can we not just not swap to a new item during onAdded if we already have one attached to the skill? And then during removeSelf swap to a new item if ours isn't equipped?